source/server/hot_restart_impl.cc

```diff
@@ -479,5 +479,15 @@ void HotRestartImpl::shutdown() { socket_event_.reset(); }
 
 std::string HotRestartImpl::version() { return shmem_.version() + "." + stats_set_.version(); }
 
+// Called from envoy --hot-restart-version -- needs to instantiate a RawStatDataSet so it
+// can generate the version string.
+std::string HotRestartImpl::hotRestartVersion(size_t max_num_stats, size_t max_stat_name_len) {
+  const SharedMemoryHashSetOptions options = sharedMemHashOptions(max_num_stats);
+  size_t bytes = RawStatDataSet::numBytes(options);
+  std::unique_ptr<uint8_t> mem_buffer_for_dry_run_(new uint8_t[bytes]);
+  RawStatDataSet stats_set(options, true /* init */, mem_buffer_for_dry_run_.get());
+  return SharedMemory::version(max_num_stats, max_stat_name_len) + "." + stats_set.version();
```

Comment:
nit: somehow this part and `HotRestartImpl::version()` would be combined to avoid an oops in the future, or I would at least drop a small comment in both functions.

Reply:
That was bugging me too.  Fixed.